xamarin/Xamarin.Forms
# Cache cell sizes after resize from estimates

### Description of Change ###

CollectionView on iOS utilizes the UICollectionViewFlowLayout's EstimatedItemSize property to allow virtualization; for items which have not yet been realized, the EstimatedItemSize is used for calculating properties like the UICollectionView's content size (for scrollbars, etc). 

When the layout is invalidated, this size is also used for newly inserted items (which have not yet had their actual sizes determined by a cell) and for any items invalidated by insertion/removal. For single DataTemplate scenarios, this is generally fine, since the estimated size calculated from the first item is likely to be close to or exactly the size of subsequent items. 

These changes now cache the size of each cell as they are updated from the initial estimated size, and reuses the cached values during future operations.

### Issues Resolved ### 

- fixes #10842
- fixes #10625 

### API Changes ###
 
None

### Platforms Affected ### 

- iOS

### Behavioral/Visual Changes ###

None

### Before/After Screenshots ### 

Not applicable

### Testing Procedure ###

Control Gallery -> CollectionView Gallery -> DataTemplate Galleries -> Varied Size Data Templates

Add/Insert/Remove items - the animation should be smooth, and no items other than the ones being inserted/removed should change size.

Control Gallery -> CollectionView Gallery -> ItemSize Galleries -> Chat Example

Add items - the animation should be smooth, and items should not jump around

### PR Checklist ###
